client/src/worldgen.rs

```diff
@@ -384,21 +423,51 @@ impl Surface {
     }
 }
 
-fn trilerp<N: na::RealField>(
+fn triserp<N: na::RealField>(
     &[v000, v001, v010, v011, v100, v101, v110, v111]: &[N; 8],
     t: na::Vector3<N>,
+    threshold1: N,
+    threshold2: N,
 ) -> N {
-    fn lerp<N: na::RealField>(v0: N, v1: N, t: N) -> N {
-        v0 * (N::one() - t) + v1 * t
+    fn serp<N: na::RealField>(v0: N, v1: N, t: N, threshold1: N, threshold2: N) -> N {
+        let out: N;
+        if t < threshold1 {
+            out = v0;
+        } else if t < threshold2 {
+            let s = (t - threshold1) / (threshold2 - threshold1);
+            // S-shaped curve that in practice is barely different to a straight line
```

Comment:
If "barely different" is true, then we shouldn't waste precious CPU time and code readability budget on it.

Reply:
Good point. serp is now piecewise linear.